'make' fails in setting up virtualenv and installing pbtranscript tofu

Hello everyone, 

I am following the description of installing pbtranscript tofu (https://github.com/PacificBiosciences/cDNA_primer/wiki/Setting-up-virtualenv-and-installing-pbtranscript-tofu#installing-tofu)

The fist steps seem all successful, but then in the make step, I get the error below.

My initial steps were:
```
sudo pip install virtualenv

/fastdisk1/pacbio/smrt/smrtanalysis/smrtcmds/bin/smrtshell

SEYMOUR_HOME=/fastdisk1/pacbio/smrt/smrtanalysis/install/smrtanalysis_2.3.0.140936

export  VENV_TOFU=/fastdisk3/thomas/pacbio/VENV_TOFU

virtualenv --system-site-packages -p /fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/bin/python $VENV_TOFU

source $VENV_TOFU/bin/activate

(VENV_TOFU)(smrtshell-2.3.0) [smrt@hal9000 ~]$ pip install bx-python

(VENV_TOFU)(smrtshell-2.3.0) [smrt@hal9000 pacbio]$ git clone https://github.com/PacificBiosciences/cDNA_primer.git


cd cDNA_primer/pbtranscript-tofu/pbtranscript/pbtools/pbtranscript/branch/C/

(VENV_TOFU)(smrtshell-2.3.0) [smrt@hal9000 C]$ python setup2.py build_ext --inplace

cd ../../../../

make
```
I hope someone can see what is going wrong. Thank you in advance!
Thomas

Output from make command
----------------------------------------------------------------------------------------------------------------------
```
(VENV_TOFU)(smrtshell-2.3.0) [smrt@hal9000 pbtranscript]$ make
python setup.py build --executable="/usr/bin/env python"
running build
running build_py
creating build
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/pbtools
copying pbtools/__init__.py -> build/lib.linux-x86_64-2.7/pbtools
creating build/lib.linux-x86_64-2.7/pbtools/pbtranscript
copying pbtools/pbtranscript/PBTranscriptOptions.py -> build/lib.linux-x86_64-2.7/pbtools/pbtranscript
copying pbtools/pbtranscript/tmp.py -> build/lib.linux-x86_64-2.7/pbtools/pbtranscript
copying pbtools/pbtranscript/tofu_wrap.py -> build/lib.linux-x86_64-2.7/pbtools/pbtranscript
...
copying pbtools/pbtranscript/data/gcon_out.fa -> build/lib.linux-x86_64-2.7/pbtools/pbtranscript/data
running build_ext
Traceback (most recent call last):
  File "setup.py", line 76, in <module>
    'bx-python'
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/core.py", line 152, in setup
    dist.run_commands()
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/command/build.py", line 127, in run
    self.run_command(cmd_name)
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/fastdisk3/thomas/pacbio/VENV_TOFU/lib/python2.7/site-packages/setuptools/command/build_ext.py", line 49, in run
    _build_ext.run(self)
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/Cython/Distutils/build_ext.py", line 163, in run
    _build_ext.build_ext.run(self)
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/command/build_ext.py", line 339, in run
    self.build_extensions()
  File "/fastdisk1/pacbio/smrt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/Cython/Distutils/build_ext.py", line 171, in build_extensions
    self.build_extension(ext)
  File "/fastdisk3/thomas/pacbio/VENV_TOFU/lib/python2.7/site-packages/setuptools/command/build_ext.py", line 169, in build_extension
    ext._convert_pyx_sources_to_lang()
AttributeError: Extension instance has no attribute '_convert_pyx_sources_to_lang'

make: *** [build] Error 1
```

I am also having this exact compiling issue running Ubuntu 14.04, and smrtanalysis 2.3.0 patch5. Installing and reinstalling setuptools (v19.4) and bx-python (0.7.3). Even tried downgrading setuptools to v18.3.2 and v18.0.  No luck. 

>(vtofu)(smrtshell-2.3.0) smrtcore@CRI-13-79:~/cDNA_primer/pbtranscript-tofu/pbtranscript$ make
python setup.py build --executable="/usr/bin/env python"
running build
running build_py
running build_ext
Traceback (most recent call last):
  File "setup.py", line 90, in <module>
    'bx-python'
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/core.py", line 152, in setup
    dist.run_commands()
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/command/build.py", line 127, in run
    self.run_command(cmd_name)
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "build/bdist.linux-x86_64/egg/setuptools/command/build_ext.py", line 49, in run
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/Cython/Distutils/build_ext.py", line 163, in run
    _build_ext.build_ext.run(self)
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/distutils/command/build_ext.py", line 339, in run
    self.build_extensions()
  File "/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/Cython/Distutils/build_ext.py", line 171, in build_extensions
    self.build_extension(ext)
  File "build/bdist.linux-x86_64/egg/setuptools/command/build_ext.py", line 169, in build_extension
AttributeError: Extension instance has no attribute '_convert_pyx_sources_to_lang'
make: *** [build] Error 1

##### Added the block of code to the top of 'setup.py'

#from setuptools import setup, find_packages
#from distutils.extension import Extension

###Added
try:
    from Cython.Distutils import build_ext
    from setuptools import setup, Extension
    HAVE_CYTHON = True
except ImportError as e:
    HAVE_CYTHON = False
    warnings.warn(e.message)
    from distutils.core import setup, Extension
    from distutils.command import build_ext
###end added block of code

import sys
import numpy as np

__author__ = "jdrake|etseng|yli@pacificbiosciences.com"
version = "2.2.3"
...

I end up with a different error

>(vtofu)(smrtshell-2.3.0) smrtcore@CRI-13-79:~/cDNA_primer/pbtranscript-tofu/pbtranscript$ make
python setup.py build --executable="/usr/bin/env python"
Traceback (most recent call last):
  File "setup.py", line 86, in <module>
    **packages=find_packages()**,
NameError: name 'find_packages' is not defined
make: *** [build] Error 1

Commenting out like 86 or adding find_packages back into the import line causes the original error.  Any other suggestions? 

Thanks,
Brian

Hi pb-cdunn,

I just saw your new comment.  This is replying to the previous:

The virtualenv was created using, "/opt/smrtanalysis/current/redist/python2.7/bin/python"
> python /home/smrtcore/virtualenv-13.1.2/virtualenv.py --system-site-packages -p /opt/smrtanalysis/current/redist/python2.7/bin/python $VENV_TOFU

#### Once in the activated env:

>(vtofu)(smrtshell-2.3.0) smrtcore@CRI-13-79:~$ which python
/home/smrtcore/vtofu/bin/python
(vtofu)(smrtshell-2.3.0) smrtcore@CRI-13-79:~$ python --version
Python 2.7.3

#### I noticed the bug with setup2.py:
>(vtofu)(smrtshell-2.3.0) smrtcore@CRI-13-79:~/cDNA_primer/pbtranscript-tofu/pbtranscript/pbtools/pbtranscript/branch/C$
python setup2.py build_ext --inplace
running build_ext
skipping 'c_branch.c' Cython extension (up-to-date)
building 'c_branch' extension
creating build
creating build/temp.linux-x86_64-2.7
gcc -pthread -fno-strict-aliasing -fPIC -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/smrtanalysis/current/redist/python2.7/include/python2.7 -c c_branch.c -o build/temp.linux-x86_64-2.7/c_branch.o
c_branch.c:346:31: fatal error: numpy/arrayobject.h: No such file or directory
 #include "numpy/arrayobject.h"
                               ^
compilation terminated.
error: command 'gcc' failed with exit status 1

#### Found a stack overflow thread about a similar numpy problem here:
http://stackoverflow.com/questions/2379898/make-distutils-look-for-numpy-header-files-in-the-correct-place

#### So I added a few new lines to 'setup2.py' and it got me further.  

    from distutils.core import setup
    from distutils.extension import Extension
    from Cython.Distutils import build_ext
    import numpy as np  ### <--- Added

    ext_modules = [#Extension("BioReaders", ["BioReaders.pyx"]), \
            Extension("c_branch", ["c_branch.pyx"]),\
            Extension("modified_bx_intervals.intersection_unique", ["modified_bx_intervals/intersection_unique.pyx"])]

    setup(
            name = 'c_cDNA_primer',
            cmdclass = {'build_ext': build_ext},
            include_dirs = [np.get_include()], ### <--- added
            ext_modules = ext_modules,
            author_email='etseng@pacificbiosciences.com',
            author='Elizabeth Tseng')
...

#### Then my output left several warnings about unused functions and depreciated NumPy APIs in use. Maybe this was the source of the problem? :
>(vtofu)(smrtshell-2.3.0) smrtcore@CRI-13-79:~/cDNA_primer/pbtranscript-tofu/pbtranscript/pbtools/pbtranscript/branch/C$
python setup2.py build_ext --inplace
running build_ext
skipping 'c_branch.c' Cython extension (up-to-date)
building 'c_branch' extension
gcc -pthread -fno-strict-aliasing -fPIC -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include -I/opt/smrtanalysis/current/redist/python2.7/include/python2.7 -c c_branch.c -o build/temp.linux-x86_64-2.7/c_branch.o
In file included from /opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/ndarraytypes.h:1728:0,
                 from /opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/ndarrayobject.h:17,
                 from /opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/arrayobject.h:15,
                 from c_branch.c:346:
/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/npy_deprecated_api.h:11:2: warning: #warning "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"
  ^
In file included from /opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/ndarrayobject.h:26:0,
                 from /opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/arrayobject.h:15,
                 from c_branch.c:346:
/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/__multiarray_api.h:1594:1: warning: '_import_array' defined but not used [-Wunused-function]
 _import_array(void)
 ^
In file included from /opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/ufuncobject.h:311:0,
                 from c_branch.c:347:
/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include/numpy/__ufunc_api.h:236:1: warning: '_import_umath' defined but not used [-Wunused-function]
 _import_umath(void)
 ^
gcc -pthread -shared -static-libgcc -L/home/vagrant/depot/software/smrtanalysis/prebuilt.out/zlib/zlib-1.2.5/centos-5/lib build/temp.linux-x86_64-2.7/c_branch.o -o /home/smrtcore/cDNA_primer/pbtranscript-tofu/pbtranscript/pbtools/pbtranscript/branch/C/c_branch.so
skipping 'modified_bx_intervals/intersection_unique.c' Cython extension (up-to-date)
building 'modified_bx_intervals.intersection_unique' extension
creating build/temp.linux-x86_64-2.7/modified_bx_intervals
gcc -pthread -fno-strict-aliasing -fPIC -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/smrtanalysis/current/redist/python2.7/lib/python2.7/site-packages/numpy-1.7.1-py2.7-linux-x86_64.egg/numpy/core/include -I/opt/smrtanalysis/current/redist/python2.7/include/python2.7 -c modified_bx_intervals/intersection_unique.c -o build/temp.linux-x86_64-2.7/modified_bx_intervals/intersection_unique.o
gcc -pthread -shared -static-libgcc -L/home/vagrant/depot/software/smrtanalysis/prebuilt.out/zlib/zlib-1.2.5/centos-5/lib build/temp.linux-x86_64-2.7/modified_bx_intervals/intersection_unique.o -o /home/smrtcore/cDNA_primer/pbtranscript-tofu/pbtranscript/pbtools/pbtranscript/branch/C/modified_bx_intervals/intersection_unique.so

Thanks for your help,
Brian

--- a/pbtranscript-tofu/pbtranscript/.gitignore
+++ b/pbtranscript-tofu/pbtranscript/.gitignore
@@ -1,0 +1,5 @@
+*.egg-info/
+build/
+.eggs/
+*.egg
+*.so

--- a/pbtranscript-tofu/pbtranscript/pbtools/pbtranscript/branch/C/setup2.py
+++ b/pbtranscript-tofu/pbtranscript/pbtools/pbtranscript/branch/C/setup2.py
@@ -1,18 +1,16 @@
-from distutils.core import setup
-from distutils.extension import Extension
-from Cython.Distutils import build_ext
+from setuptools import setup, Extension
 import numpy
 
-ext_modules = [#Extension("BioReaders", ["BioReaders.pyx"]), \
+ext_modules = [
+        #Extension("BioReaders", ["BioReaders.pyx"]),
         Extension("c_branch", ["c_branch.pyx"],
-            include_dirs=[numpy.get_include()]),\
-        Extension("modified_bx_intervals.intersection_unique", ["modified_bx_intervals/intersection_unique.pyx"])]
-
+            include_dirs=[numpy.get_include()]),
+        Extension("modified_bx_intervals.intersection_unique", ["modified_bx_intervals/intersection_unique.pyx"]),
+]
 setup(
 		name = 'c_cDNA_primer',
-        cmdclass = {'build_ext': build_ext},
+        #cmdclass = {'build_ext': build_ext},
 		ext_modules = ext_modules,
         author_email='etseng@pacificbiosciences.com',
         author='Elizabeth Tseng'
 )
-

--- a/pbtranscript-tofu/pbtranscript/setup.py
+++ b/pbtranscript-tofu/pbtranscript/setup.py
@@ -67,8 +67,8 @@ setup(
                    'data/gcon_out.fa']},
     packages=find_packages(),
     install_requires=[
-#        'pbcore >= 0.6.3',
-#        'bx-python',
+        'pbcore >= 0.6.3',
+        'bx-python',
 #        'setuptools_cython',
         ],
     zip_safe=False,

--- a/pbtranscript-tofu/pbtranscript/setup.py
+++ b/pbtranscript-tofu/pbtranscript/setup.py
@@ -1,17 +1,16 @@
-from setuptools import setup, find_packages
-from distutils.extension import Extension
+from setuptools import setup, Extension, find_packages
 import sys
-import numpy as np
-
+import numpy
 
 __author__ = "jdrake|etseng|yli@pacificbiosciences.com"
 version = "2.2.3"
 
-if 'setuptools.extension' in sys.modules:
-    m = sys.modules['setuptools.extension']
-    m.Extension.__dict__ = m._Extension.__dict__
+#if 'setuptools.extension' in sys.modules:
+#    m = sys.modules['setuptools.extension']
+#    m.Extension.__dict__ = m._Extension.__dict__
 
-ext_modules = [Extension("pbtools.pbtranscript.findECE",
+ext_modules = [
+                Extension("pbtools.pbtranscript.findECE",
                          ["pbtools/pbtranscript/ice/C/findECE.c"]),
                 Extension("pbtools.pbtranscript.io.c_basQV",
                          ["pbtools/pbtranscript/ice/C/c_basQV.cpp"], language="c++"),
@@ -20,24 +19,23 @@ ext_modules = [Extension("pbtools.pbtranscript.findECE",
 #                Extension("pbtools.pbtranscript.c_Prob", 
 #                         ["pbtools/pbtranscript/ice/C/ProbHandler.cpp"], language="c++"),
                 Extension("pbtools.pbtranscript.ice.ProbModel",
-                    ["pbtools/pbtranscript/ice/C/ProbModel.cpp"], language="c++"),
+                         ["pbtools/pbtranscript/ice/C/ProbModel.cpp"], language="c++"),
                 Extension("pbtools.pbtranscript.BioReaders",
                          ["pbtools/pbtranscript/io/C/BioReaders.c"]),
                 Extension("pbtools.pbtranscript.modified_bx_intervals.intersection_unique",
-                    ["pbtools/pbtranscript/branch/C/modified_bx_intervals/intersection_unique.c"]),
+                         ["pbtools/pbtranscript/branch/C/modified_bx_intervals/intersection_unique.c"]),
                 Extension("pbtools.pbtranscript.c_branch", 
                          ["pbtools/pbtranscript/branch/C/c_branch.c"]), 
-              ]
-
+]
 setup(
-#    setup_requires=['setuptools_cython'],
+    setup_requires=['setuptools_cython'],
     name = 'pbtools.pbtranscript',
     version=version,
     author='Pacific Biosciences',
     author_email='devnet@pacificbiosciences.com',
     license='LICENSE.txt',
     ext_modules = ext_modules,
-    include_dirs = [np.get_include()],
+    include_dirs = [numpy.get_include()],
     scripts=['pbtools/pbtranscript/pbtranscript.py',
              'pbtools/pbtranscript/collapse_isoforms_by_sam.py',
              'pbtools/pbtranscript/fusion_finder.py',
@@ -47,12 +45,10 @@ setup(
              'pbtools/pbtranscript/ice_pbdagcon.py',
              'pbtools/pbtranscript/ice_quiver.py',
              'pbtools/pbtranscript/ice_fa2fq.py',
-            'pbtools/pbtranscript/picking_up_ice.py',
-            'pbtools/pbtranscript/cleanup_ice.py',
-            'pbtools/pbtranscript/counting/chain_samples.py'
+             'pbtools/pbtranscript/picking_up_ice.py',
+             'pbtools/pbtranscript/cleanup_ice.py',
+             'pbtools/pbtranscript/counting/chain_samples.py'
         ],
-
-
     entry_points={'console_scripts': [
         'fasta_splitter.py = pbtools.pbtranscript.io.FastaSplitter:main',
         'filter_sam.py = pbtools.pbtranscript.io.filter_sam:main',
@@ -70,11 +66,12 @@ setup(
                    'data/gcon_in.fa',
                    'data/gcon_out.fa']},
     packages=find_packages(),
-    zip_safe=False,
     install_requires=[
-        'pbcore >= 0.6.3',
-        'bx-python'
-        ]
+#        'pbcore >= 0.6.3',
+#        'bx-python',
+#        'setuptools_cython',
+        ],
+    zip_safe=False,
     )
 
 #'pbtools/pbtranscript/io/FastaSplitter.py',

